vendor/net_gearman/Net/Gearman/AtomWorker.php

```diff
@@ -0,0 +1,152 @@
+     */
+    public function beginWork($monitor = null)
+    {
+        if (!is_callable($monitor)) {
+            $monitor = [$this, 'stopWork'];
+        }
+        $write = null;
+        $except = null;
+        $working = true;
+        $lastJob = time();
+        $retryTime = 5;
+        while ($working) {
+            $sleep = true;
+            $currentTime = time();
+            foreach ($this->conn as $server => $socket) {
+                $worked = false;
+
+                try {
+                    $worked = $this->doWork($socket);
+                } catch (Net_Gearman_Exception $e) {
+                    unset($this->conn[$server]);
+                    $this->retryConn[$server] = $currentTime;
+                }
+                if ($worked) {
+                    $lastJob = time();
+                    $sleep = false;
+                }
+            }
+            $idle = false;
+            if ($sleep && count($this->conn)) {
+                foreach ($this->conn as $socket) {
+                    Net_Gearman_Connection::send($socket, 'pre_sleep');
+                }
+                $read = $this->conn;
+                @socket_select($read, $write, $except, $this->socket_timeout);
+                $idle = (0 == count($read));
+            }
+            $retryChange = false;
+            foreach ($this->retryConn as $s => $lastTry) {
+                if (($lastTry + $retryTime) < $currentTime) {
+                    try {
+                        $conn = Net_Gearman_Connection::connect($s);
+                        $this->conn[$s] = $conn;
+                        $retryChange = true;
+                        unset($this->retryConn[$s]);
+                        Net_Gearman_Connection::send($conn, 'set_client_id', ['client_id' => $this->id]);
+                    } catch (Net_Gearman_Exception $e) {
+                        $this->retryConn[$s] = $currentTime;
+                    }
+                }
+            }
+            if (0 == count($this->conn)) {
+                // sleep to avoid wasted cpu cycles if no connections to block on using socket_select
+                sleep(1);
+            }
+            if (true === $retryChange) {
+                // broadcast all abilities to all servers
+                foreach ($this->abilities as $ability => $timeout) {
+                    $this->addAbility(
+                        $ability, $timeout, $this->initParams[$ability]
+                    );
+                }
+            }
+            if (true == call_user_func($monitor, $idle, $lastJob)) {
+                $working = false;
+            }
+
+            if ($this->maxJobCountReached()) {
+                $working = false;
+            }
+        }
+    }
+
+    protected function maxJobCountReached()
+    {
+        if ($this->maxJobCount <= 0) {
+            return false;
+        }
+
+        if ($this->jobsCompleted >= $this->maxJobCount) {
```

Comment:
This condition and the above could be combined into:
```
if ($this->maxJobCount > 0 && $this->jobsCompleted >= $this->maxJobCount) {
```

Reply:
I agree.